tomcat/src/main/java/org/apache/coyote/http11/Http11Processor.java

```diff
@@ -76,20 +76,24 @@ private List<String> readHeaders(final BufferedReader bufferedReader) throws IOE
     }
 
     private String readRequestBody(final List<String> headers, final BufferedReader bufferedReader) throws IOException {
-        final StringBuilder stringBuilder = new StringBuilder();
-        if (headers.contains("Content-Length")) {
-            addRequestBodyLine(bufferedReader, stringBuilder);
+        final int contentLength = getContentLength(headers);
+        if (contentLength != 0) {
+            final char[] buffer = new char[contentLength];
+            bufferedReader.read(buffer, 0, contentLength);
+            return new String(buffer);
         }
-        return stringBuilder.toString();
+        return "";
     }
 
-    private void addRequestBodyLine(final BufferedReader bufferedReader, final StringBuilder requestBody)
-            throws IOException {
-        for (String requestBodyLine = bufferedReader.readLine();
-             requestBodyLine != null && !requestBodyLine.equals("");
-             requestBodyLine = bufferedReader.readLine()) {
-            requestBody.append(requestBodyLine);
-        }
+    private int getContentLength(final List<String> headers) {
+        return headers.stream()
+                .filter(header -> header.startsWith(HttpHeaders.CONTENT_LENGTH))
+                .findAny()
+                .map(contentLengthHeader -> {
+                    final int headerKeyValueSeparatorIndex = contentLengthHeader.indexOf(":");
```

Comment:
```suggestion
                    final int headerSeparatorIndex = contentLengthHeader.indexOf(":");
```
헤더에는 key value 형식밖에는 없으니 축약해서 쓰는건 어떤가요?

Reply:
headerSeparator라고 했을 경우 헤더끼리의 구분자를 뜻하는 것 같아 `키와 밸류를 나누는 구분자`라는 의미를 부여하고 싶었어요. 다만 해당 로직이 작성되는 곳은 명백히 Header에 관한 공간이기 때문에 header라는 네이밍은 빼도 될 것 같네요. `keyValueSeparatorIndex`로 변경했습니다!